src/writer/serve.py

```diff
@@ -122,6 +139,7 @@ async def lifespan(asgi_app: FastAPI):
     """
     app.state.writer_app = True
     app.state.app_runner = app_runner
+    app.state.job_vault = JobVault()
```

Comment:
Job Vault will be used to store results of async jobs.

A basic dict-based structure for single-node deployments. There'll be an option to use Redis for bigger deployments (we can then use this same Redis to do other interesting stuff).

Reply:
I have one concern. I don't know how to save the result of the job in the job vault. 

If I understand correctly the goal is to create an API endpoint for triggering a job decoupled from a user session. WF retrieves it and executes it on an `Event handler` of the `AppProcess` 

In this case, the job vault will serve as memory. When the job is finished, it will allow the user to retrieve the result. It is only used for "synchronous" jobs for which we are interested in the result.

My concern is the job vault is in the primary process, the thread pool that run a job in async in WF is in the secondary process. You can write something into the job vault from the secondary process. I have tried to explain what I have understand in the schema below.

![image](https://github.com/user-attachments/assets/b0b6255b-909b-4ae6-9f5c-cc7723db224e)

Reply:
Ok, you rely on ``loop.create_task``. That was the part I wasn't sure. Looks ok for me then.

![image](https://github.com/user-attachments/assets/57d79dde-53e9-47f4-a93c-a14633d5ef5b)

```python
loop = asyncio.get_running_loop()
        task = loop.create_task(app_runner.handle_event(
            session_id, WriterEvent(
                type="wf-builtin-run",
                isSafe=True,
                handler=f"$runWorkflow_{workflow_key}"
            )))
```